lib/rack/utils.rb

```diff
@@ -248,13 +248,20 @@ def add_cookie_to_header(header, key, value)
           rfc2822(value[:expires].clone.gmtime) if value[:expires]
         secure = "; secure"  if value[:secure]
         httponly = "; HttpOnly" if (value.key?(:httponly) ? value[:httponly] : value[:http_only])
-        first_party = "; First-Party" if value[:first_party]
+        same_site = case (value[:first_party] || value[:same_site]).to_s.downcase
+        when "lax"
+          "; SameSite=Lax"
+        when "strict"
+          "; SameSite=Strict"
+        when "true"
+          "; SameSite"
```

Comment:
Ah yes. It came to mind as I read the spec:

> If the "SameSite" attribute has no value, if the value is "Strict", *or if the value is invalid*

Which struck me as a bit puzzling. Whether that's meant to cover implementor mistakes by failing back to a strict default or whether it's meant to act as forward compatibility with new SameSite values.

We could be better off passing through the provided value verbatim. The spec is clear that attribute name and value matching is case insensitive, so passing any variety of lax/strict value would Just Work, and what we see as invalid today may be valid tomorrow, in which case we wouldn't want to be dropping the value entirely, losing strictness.